src/app/domain/usecases/HintChampion.ts

```diff
@@ -0,0 +1,47 @@
+import { IClientStorage } from 'src/app/infra/client-storage/iclient-storage';
+import ServiceFactory from 'src/app/infra/factory/ServiceFactory';
+import ChampionService from 'src/app/services/ChampionService';
+import { SafeAny } from 'src/safeAny';
+import { ChampionEntity } from '../entities/champion/champion-entity';
+
+export default class HintChampion {
+  championService: ChampionService;
+
+  constructor(
+    readonly localStorageService: IClientStorage,
+    readonly serviceFactory: ServiceFactory
+  ) {
+    this.championService = serviceFactory.createChampionService();
+  }
+
+  isSameChampion(championName: string, lastChampions: SafeAny) {
+    return championName === lastChampions.pickedChampion;
+  }
+
+  async execute(input: Input): Promise<Output> {
+    const lastChampions = this.localStorageService.get('champions');
+    const champions = await this.championService.list();
+    const pickedChampion = champions.filter((champion) =>
+      this.isSameChampion(champion.name, lastChampions)
+    )[0];
```

Comment:
I have a fear that this returns an empty value sometime. We could add an if to check if this filter's result is an array and have values.